src/controller/middleware/validation.js

```diff
@@ -0,0 +1,19 @@
+exports.signupValidation = (req, res, next) => {
+  const { userName, password, confirmPassword } = req.body;
+  if (!userName.trim() || !password.trim() || !confirmPassword.trim()) {
+    return res.status(200).send(JSON.stringify('يجب ملئ جميع الحقول'));
+  }
+  if (userName.length < 4 || userName.length > 20) {
+    return res.status(200).send(JSON.stringify('اسم المستخدم يجب ألا يزيد عن 20 وألا يقل عن 4 أحرف'));
+  }
+  if (userName.includes(' ')) {
+    return res.status(200).send(JSON.stringify('اسم المستخدم يجب ألا يحتوي على مسافات'));
+  }
+  if (password.length < 3) {
+    return res.status(200).send(JSON.stringify('كلمة المرور يجب ألا يقل عن 3 أحرف/أرقام'));
+  }
+  if (password.trim() !== confirmPassword.trim()) {
+    return res.status(200).send(JSON.stringify('يجب تطابق كلمتي المرور'));
```

Comment:
@AymanAlqoqa This is really good.  One idea I had is to change all of the status codes from `200` (which means success) to something that indicates to the client that there was an error.  How about `400` (bad request)?

In general, 2xx codes means "everything worked fine", 4xx codes mean "error caused by the user", and 5xx codes mean "error caused by the server".  I usually try to pick meaningful error codes because then the frontend code can turn that into something helpful to the user.

Reply:
thanks a lot for your helpful feedback. I agree with you status code should be meaningful.
I will modify it as soon as possible
thanks